best-practices/tech_lead.md

```diff
@@ -0,0 +1,318 @@
+
+# It's About Team Achievement
+
+"excelling in delivery and execution doesn't just mean being able to do those things yourself, but being able to coordinate, communicate, plan, and delegate effectively." - https://blog.coleadership.com/what-does-it-take-to-be-a-successful-tech-lead/
+
+TL is a full-time job; I don't believe you can be a good TL and a developer at the same time because being TL takes at least 70% of work time, and the remaining time is usually broken up into a few minutes here and there.
+
+## It's about OTHER people
+
+The TL role is to help the _other_ devs be as productive as possible on sprint and work-cycle goals.  The TL is the interface to the info the team needs to work effectively.
+- knock down roadblocks, or help find another way around
+- make sure PRs are reviewed promptly, and with an eye to quality and coding goals (below)
+- ensure team awareness of current priorities, which do shift
+- keep work on track
+- try to anticipate gotchas and address them
+
+The TL may not be the expert in the tech
+- use the team ... tap into expert knowledge wherever it resides
+
+The TL may not be able to code much
+- more devs --> more TL time needed coordinating
+- never take blocking tickets
+  - it's more important for you to be available for pairing, consulting with PO, ticket writing, board grooming, etc.
+- take tiny tickets so they won't languish half done
+- take unfun tickets so other devs can be happier
+  - e.g. review who has access to endpoints: this can be done in small pieces, and no one will be excited to do it
+
+_“What am I uniquely capable of doing?”_ - do these things; leave the rest on the board for others.  As TL, you are in a key position for information flow. Honor that via ticket writing, board grooming, pairing, etc.
+
+It helps to know who you are, what your skills and talents are, and to ask for help when you need it.
+
+## Project Management Has to Be Done
+
+... and team members are happier if it is done well.
+
+### It's About Process
+There needs to be a transparent process that:
+- documents the work cycle and sprint goals for easy lookup (e.g epics, milestones)
+- enables devs to easily choose the highest priority work for the sprint and work cycle (e.g ready column)
+- keeps all devs actively engaged in working towards sprint and work cycle goals (e.g. population of ready column, easy way for devs working on a ticket to surface problems and get resolution quickly)
+- makes it easy for the Tech Lead (TL), the Product Owner (PO) and management to see which work is in progress and how it relates to the goals (the board as a whole)
+
+If you're TL, and the work-cycle PO is mostly unavailable, and there's no appointed scrum master ... project management still needs to be done.  You don't have to do everything yourself ... but you need to ensure it all gets done.
```

Comment:
I am not sure all projects have a scrummaster, so I guess the question of who does the logistical work of scheduling and facilitating meetings in that case is up for discussion (partly done by TL and partly done by PO?)

Reply:
my main point is really that SOMEONE has to do it.  If no one does it, we all suffer.   Hence my statement that you can farm it out, but the TL is on the team and the PO usually isn't, so the TL is more likely to need to ensure it happens.

Reply:
i finessed the wording a bit - thanks for the feedback.